build.sbt

```
@@ -47,6 +47,10 @@ ThisBuild / printSparkScalaVersion := {
 
 Test / parallelExecution := false
 
+// Only apply scalafmt to files that differ from master (i.e. files changed in the feature branch or so),
+// not on the whole repository.
+scalafmtFilter.withRank(KeyRanks.Invisible) := "diff-ref=origin/master"
```

This works here, but would not work in projects that have master & develop.
This would then need to be something like 

```suggestion
val branchName = if (Process("git branch --show-current").lineStream.head.equals("develop")) {
  "master"
} else {
  "develop"
}
scalafmtFilter.withRank(KeyRanks.Invisible) := s"diff-ref=origin/${branchName}"
```

---

Or even better. Getting a parent branch name, regardless of the flow

```shell
git show-branch -a \
| grep '\*' \
| grep -v `git rev-parse --abbrev-ref HEAD` \
| head -n1 \
| sed 's/.*\[\(.*\)\].*/\1/' \
| sed 's/[\^~].*//'
```

---

Yeah I was thinking about something like this - the 'base' branch; unfortunately there isn't a single command that could be used.

I like both of your proposals, but the second one is more precise. Just a minor improvement, perhaps I'd use `git branch --show-current` instead of `git rev-parse --abbrev-ref HEAD` since it's simpler

---

Thank you!

---

I take it back, it works on my machine but not in the CI. There is some problem with grep. I'll take a look.

---

Refactored, please check the most recent commit.